### ISSUE TITLE
negative loss and failed to load model for retraining

### ISSUE THREAD

**Issue by benlaitang** [NVIDIA/waveglow#15]
I encountered 2 issues when following the work.
The first one is when I trained the model from the scratch. I got some negative loss. I read your paper and think this case is normal, right?
However,  after 7 epochs about 22000 iterations, I synthesized the audio,  got some white noise. Are there any mistakes?  How many epochs and iterations do you train?
The second one is I retrained the model from your "waveglow_old.pt ",  when loading the model but lost "iteration", "optimizer" and "learning_rate". When training, I got "tensor(nan., device='cuda:0')" and loss is Nan.

Can you help me or give some advices?

**Comment by chaiyujin** [comment]
I moved the waveglow into my training framework and trained it with DataParallel from scratch. I also got negative loss and 'nan' during training.
![waveglow-nll](https://user-images.githubusercontent.com/10378898/48395280-c51a4300-e751-11e8-9c3a-219b7b041a70.png)

Besides, the infered samples even reach 1e17.
![waveglow-wav](https://user-images.githubusercontent.com/10378898/48395287-c9def700-e751-11e8-9876-0fcdf3b0d56b.png)

**Comment by benlaitang** [comment]
@chaiyujin so, what your  training framework?  retrain from the provided model, did it work? I am lost your words

**Comment by chaiyujin** [comment]
@benlaitang Sorry about my english. I have my own training framework.  I trained glow from scratch. Never test fine-tuning from provided model.

**Comment by wenyong-h** [comment]
I got exactly the same problem.

**Comment by benlaitang** [comment]
@wenyong-h did you solve the problem?

**Comment by wenyong-h** [comment]
No, I'm training from scratch now.

**Comment by rafaelvalle** [comment]
1) Yes, the loss should be negative. We trained the model for 580 iterations with batch size 24. 22k iters with batch size 3 is probably not enough to produce intelligible speech.
2) The second model is shared for inference only, not resuming training.

**Comment by benlaitang** [comment]
@rafaelvalle thanks a lot. I will try from scratch again.

**Comment by rafaelvalle** [comment]
Closing. Please re-open if necessary.

**Comment by yoyololicon** [comment]
@chaiyujin Did you solve the issue of `nan` loss? Because I encounter similar issue. My training curve is something like these:
![2018-11-30 08_36_14-tensorboard](https://user-images.githubusercontent.com/17811395/49261080-8be51100-f47b-11e8-9a48-984d6fdbd15f.png)
![2018-11-29 21_44_22-tensorboard](https://user-images.githubusercontent.com/17811395/49261107-a5865880-f47b-11e8-9beb-43a60e4cd68a.png)

**Comment by chaiyujin** [comment]
@yoyololicon 
1. When training with default model given by this repo, I encounter `NAN`.
   [Solve]: I initialize the upsample layer weight to be 1.0, bias to be 0.0. 
2. When training with multi-gpus, I encounter `NAN` again.
   [Solve]: Use one gpu, batch_size = 4

I'm not sure how to avoid `nan`. 
It works to train with inited upsample layer and on one gpu.

**Comment by scimagian** [comment]
@chaiyujin thanks for your solution. I also met the NAN problem. When I set the one GPU with batch size 1, the training loss is fine. But when I set 8 GPU by using torch.nn.parallel.data_parallel with batch size 8, I got NAN loss after a few thousand steps. I haven't found any solution to solve it.
